src/dune_engine/dep.ml

```diff
@@ -98,22 +96,23 @@ module Fact = struct
     let paths t = t.files
 
     let make ~files ~dirs =
-      let parent_dirs =
-        let f path (_ : Digest.t) acc =
-          Path.Set.add acc (Path.parent_exn path)
-        in
-        let init = Path.Map.foldi files ~init:Path.Set.empty ~f in
-        Path.Map.foldi files ~init ~f
-      in
       { files
       ; dirs
-      ; parent_dirs
       ; digest =
           Digest.generic
             ( Path.Map.to_list_map files ~f:(fun p d -> (Path.to_string p, d))
             , Path.Map.to_list_map dirs ~f:(fun p d -> (Path.to_string p, d)) )
       }
 
+    let necessary_dirs_for_sandboxing { files; dirs = _; digest = _ } =
+      let f path (_ : Digest.t) acc =
+        match Path.as_in_build_dir path with
+        | None -> acc
```

Comment:
I'm still uneasy about this. Do we expect build paths here? If yes, shouldn't they contribute to the result? If not, shouldn't we raise instead of silently proceeding?

Reply:
/cc @staronj Who is investigating a mysterious Dune bug related to this code. (Sandboxed actions failing becuase the rule directory has not been created.)